### ISSUE TITLE
insert-beginning: Files outside of the repository can be added to the repository

### ISSUE THREAD

**Issue by CervEdin** [newren/git-filter-repo#260]
It appears one can add a file outside of the repo using the [`insert-beginning`](https://github.com/newren/git-filter-repo/blob/main/contrib/filter-repo-demos/insert-beginning) demo.

Steps to reproduce
```
mkdir insert-outside && cd insert-outside
git init && git commit --allow-empty -m init
echo outside! > ../an-outside-file
insert-beginning --file ../an-outside-file
```

This will result in something like
```
$ git log -p
commit 61116276d40336ab23974003597ed167d36bf2d6 (HEAD -> main)
Author: Erik Cervin Edin <erik@cervined.in>
Date:   Tue Jun 15 16:34:50 2021 +0200

    init

diff --git a/../an-outside-file b/../an-outside-file
new file mode 100755
index 0000000..b454f01
--- /dev/null
+++ b/../an-outside-file
@@ -0,0 +1 @@
+outside!
```

Trying to undo results in
```
$ git filter-repo --paths ../an-outside-file --invert
Parsed 1 commitserror: invalid path '../an-outside-file'
fatal: Could not reset index file to revision 'HEAD'.
```
or
```
$ git filter-repo --path-rename "../an-outside-file:an-outside-file"
Error: Invalid path component '..' found in '../an-outside-file:an-outside-file'
```

It seems possible to really mess up a repository doing this.

**Comment by newren** [comment]
Wow, that looks bad!  I don't have time at the moment to investigate, but thanks for reporting.  I'll try to look at it soon.

**Comment by newren** [comment]
Fixed with ad7b5b8eedc1 (insert-beginning: add some sanity checking on the passed filename, 2024-11-23).  Also created a patch to have fast-import reject such paths; see https://github.com/gitgitgadget/git/pull/1831
